### PR TITLE
#145

### PR DESCRIPTION
**Pull request by mensfeld**
@agwozdowski  readme in progress

### CODE REVIEW

**Comment by mach-kernel** [comment]
@mensfeld awesome news! Can't wait to start using this!

Let me know if there are any other features / contributions planned, I'd love to help out with this project 😄 .

**Comment by mensfeld** [comment]
@mach-kernel thank you! Will keep that in mind. Sorry for taking that one over - I started to describe what I wanted to achieve and I've noticed that I wrote like 90% of the code. @agwozdowski  should review and hopefully merge it today. We have some incoming tasks to do - if you are interested, please drop me line @ maciej@mensfeld.pl as I would like to keep them private for now.

File: karafka.gemspec [hunk 1]
```diff
@@ -17,7 +17,7 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'bundler', '~> 1.2'
 
   spec.add_dependency 'ruby-kafka', '= 0.3.17'
-  spec.add_dependency 'sidekiq', '~> 4.2'
+  spec.add_dependency 'sidekiq', '~> 5.0'
```
Comment (maximebedard) [inline karafka.gemspec]: Is this bump completely necessary to add this feature? This forces everyone to upgrade to 5.0 without using any of the features required by sidekiq.

Comment (mensfeld) [inline karafka.gemspec]: @maximebedard  that is absolutely right. For now, we can open this dependency to support >= 4.2 and everyone should be happy - #148  . Also, I've created #147 to cover that better in the future.

Comment (mensfeld) [inline karafka.gemspec]: I should fix #148 today.

Comment (mensfeld) [inline karafka.gemspec]: @maximebedard  done :) sorry for any inconvenience  that changed caused and thank you for an inspiration for another Karafka feature!

Comment (maximebedard) [inline karafka.gemspec]: Amazing! Thanks to you, keep up your awesome work 😄 ❤️

**Pull request merged**